### PR TITLE
Use Snapkit for layout of routeSelectionView

### PR DESCRIPTION
**Pull request by Omarrasheed**
As title says. There is a looooooot of things wrong with how we set the data in this view (we legit call button.addTarget and we set the datePickerTitle from the VC that creates this view) and it completely destroys MVC model. I will address this in another PR, but I just want to get this snapkit stuff out of the way first

### CODE REVIEW

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 1]
```diff
@@ -101,7 +101,7 @@ class RouteOptionsViewController: UIViewController {
         // assume user wants to find routes that leave at current time and set datepicker accordingly
         searchTime = Date()
         if let searchTime = searchTime {
-            routeSelection.setDatepicker(withDate: searchTime, withSearchTimeType: searchTimeType)
```
Comment (Omarrasheed) [inline TCAT/Controllers/RouteOptionsViewController.swift:104]: Same here

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 2]
```diff
@@ -684,7 +678,7 @@ class RouteOptionsViewController: UIViewController {
             if searchTimeType == .leaveNow {
                 let now = Date()
                 searchTime = now
-                routeSelection.setDatepicker(withDate: now, withSearchTimeType: searchTimeType)
```
Comment (Omarrasheed) [inline TCAT/Controllers/RouteOptionsViewController.swift:687]: Same here

File: TCAT/Controllers/RouteOptionsViewController+Extensions.swift [hunk 3]
```diff
@@ -117,7 +117,7 @@ extension RouteOptionsViewController: DatePickerViewDelegate {
             buttonTapped = "Leave At Tapped"
         }
 
-        routeSelection.setDatepicker(withDate: date, withSearchTimeType: searchTimeType)
```
Comment (Omarrasheed) [inline TCAT/Controllers/RouteOptionsViewController+Extensions.swift:120]: Ignore this. It'll get moved later (probably using delegates? haven't really thought too much into it yet)

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 4]
```diff
@@ -153,13 +153,7 @@ class RouteOptionsViewController: UIViewController {
...
-        routeSelection = RouteSelectionView(frame: CGRect(x: 0, y: -12, width: view.frame.width, height: 150))
-        routeSelection.backgroundColor = Colors.white
-        var newRSFrame = routeSelection.frame
-        newRSFrame.size.height =  routeSelection.lineWidth + routeSelection.searcbarView.frame.height + routeSelection.lineWidth + routeSelection.datepickerButton.frame.height
-        routeSelection.frame = newRSFrame
-
+        routeSelection = RouteSelectionView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: 130))
         routeSelection.toSearchbar.addTarget(self, action: #selector(self.searchingTo), for: .touchUpInside)
         routeSelection.fromSearchbar.addTarget(self, action: #selector(self.searchingFrom), for: .touchUpInside)
         routeSelection.datepickerButton.addTarget(self, action: #selector(self.showDatePicker), for: .touchUpInside)
```
Comment (Omarrasheed) [inline TCAT/Controllers/RouteOptionsViewController.swift:159]: These 3 lines were what I mentioned about button.addTarget being called from the VC. I'll reorganize this in a later diff. Plz don't roast me for this

**Review by Omarrasheed — COMMENTED** [review]
Left some notes

File: TCAT/Controllers/RouteOptionsViewController+Extensions.swift (hunk 3 above)
Comment (kevinchan159) [reply TCAT/Controllers/RouteOptionsViewController+Extensions.swift:120]: can just use a `configure` function probably

File: TCAT/Controllers/RouteOptionsViewController.swift (hunk 4 above)
Comment (kevinchan159) [reply TCAT/Controllers/RouteOptionsViewController.swift:159]: BOOM ROASTED
prob just put this inside a `configure` function

**Comment by young-k** [comment]
@kevinchan159 Did you set up branch protection on this repo?

File: TCAT/Views/RouteSelectionView.swift [hunk 5]
```diff
@@ -156,139 +122,80 @@ class RouteSelectionView: UIView {
...
-        searcbarView.frame = newFrame
-    }
+    private func setupConstraints() {
+        let separatorHeight: CGFloat = 1.0
 
-    private func positionDatepickerButton(usingSearchbarView searchbarView: UIView) {
-        let oldFrame = datepickerButton.frame
-        let newFrame = CGRect(x: 0, y: searchbarView.frame.maxY + lineWidth, width: oldFrame.width, height: oldFrame.height)
+        swapButton.snp.makeConstraints { make in
+            make.size.equalTo(CGSize(width: 20, height: 25))
```
Comment (kevinchan159) [inline TCAT/Views/RouteSelectionView.swift]: can you move all these numbers into constants at the top of the function just like line 126

**Review by kevinchan159 — APPROVED** [review]
noice, just left one comment

**Comment by kevinchan159** [comment]
> @kevinchan159 Did you set up branch protection on this repo?

ye i did

**Review by young-k — APPROVED** [review]

**Pull request merged**